src/pages/en/guides/migrate-to-astro/from-nextjs.mdx

```diff
@@ -2,72 +2,924 @@
+
+When part of a [content collection](/en/guides/content-collections/), Markdown and MDX files will live in folders within `src/content/` and you will [generate those pages dynamically](/en/guides/content-collections/#generating-pages-from-content-collections).
+
+### Migrating Tests
+
+As Astro outputs raw HTML, it is possible to write end-to-end tests using the output of the build step. Any end-to-end tests written previously might work out-of-the-box, if you have been able to match the markup of your Next site. Testing libraries such as Jest and React Testing Library can be imported and used in Astro to test your React components.
+
+See Astro's [testing guide](/en/guides/testing/) for more.
+
+## Reference: Convert Next.js Syntax to Astro
+
+### Next Links to Astro
+
+Convert any Next `<Link to="">`, `<NavLink>` etc. components to HTML `<a href="">` tags. 
+
+```astro del={1} ins={2}
+<Link to="/blog">Blog</Link>
+<a href="/blog">Blog</a>
+```
+
+Astro does not use any special component for links, although you are welcome to build custom link components. You can then import and use this `<Link>` just as you would any other component.
+
+```astro title="src/components/Link.astro"
+---
+const { to } = Astro.props
+---
+<a href={to}><slot /></a>
+```
+
+### Next Imports to Astro
+
+If necessary, update any [file imports](/en/guides/imports/) to reference relative file paths exactly. This can be done using [import aliases](/en/guides/typescript/#import-aliases), or by writing out a relative path in full. 
+
+Note that `.astro` and several other file types must be imported with their full file extension.
+
+```astro title="src/pages/authors/Fred.astro"
+---
+import Card from `../../components/Card.astro`
+---
+<Card />
+```
+
+### Next Children Props to Astro
+
+Convert any instances of `{children}` to an Astro `<slot />`. Astro does not need to receive `{children}` as a function prop and will automatically render child content in a `<slot />`.
+
+```astro title="src/components/MyComponent" del={3-9} ins={11-13}
+---
+---
+export default function MyComponent(props) { 
+    return (
+      <div>
+        {props.children}
+      </div>
+    );  
+}
+
+<div>
+  <slot />
+</div>
+```
+
+React components that pass multiple sets of children can be migrated to an Astro component using [named slots](/en/core-concepts/astro-components/#named-slots).
+
+See more about [specific `<slot />` usage in Astro](/en/core-concepts/astro-components/#slots).
+
+### Next Data Fetching to Astro
+
+Convert any instances of `getStaticProps()` to `Astro.glob()` or `getCollections()` to access data from other files in your project source. To [fetch remote data](/en/guides/data-fetching/), use `fetch()`.
+
+These data requests are made in the frontmatter of the Astro component using the data, and use top-level await.
```

Comment:
This sentence felt a bit odd to me, probably because of the double "use" verb, maybe changing a bit works?
```suggestion
These data requests are made in the frontmatter of the Astro component using the data with top-level await.
```